pkg/strategy/grid/strategy.go

```diff
@@ -431,6 +461,21 @@ func (s *Strategy) Run(ctx context.Context, orderExecutor bbgo.OrderExecutor, se
 	s.Graceful.OnShutdown(func(ctx context.Context, wg *sync.WaitGroup) {
 		defer wg.Done()
 
+		if s.Persistence != nil {
+			log.Infof("backing up active orders...")
+			submitOrders := s.activeOrders.Backup()
+			snapshot := Snapshot{
+				Orders: submitOrders,
+				Position: &s.position,
+			}
+
+			if err := s.Persistence.Save(&snapshot, ID, s.Symbol, "snapshot"); err != nil {
```

Comment:
It might be better to pass ctx into the Save and Load method of Persistence.

Reply:
let's fix this later ;)